From 24fe25b10806bf9946aecea73f770a42081e6465 Mon Sep 17 00:00:00 2001
From: Florian Weimer <fweimer@redhat.com>
Date: Fri, 28 Aug 2026 10:26:07 +0200
Subject: [PATCH 31/52] UPSTREAM: iconvdata: EUC_JISX0213 decoding lacks
 pending character reset (CVE-2026-80489)

This fixes bug 34568.

Reviewed-by: Carlos O'Donell <carlos@redhat.com>
(cherry picked from commit 4dafa087ff5fe7df45bd37dc727e988da6b8c935)
---
 iconvdata/euc-jisx0213.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/iconvdata/euc-jisx0213.c b/iconvdata/euc-jisx0213.c
index 5572bbdb7b..61c22231e1 100644
--- a/iconvdata/euc-jisx0213.c
+++ b/iconvdata/euc-jisx0213.c
@@ -224,6 +224,9 @@
 	    STANDARD_FROM_LOOP_ERR_HANDLER (1);				      \
 	  }								      \
       }									      \
+    else 								      \
+      /* There was a pending character.  Clear it.  */			      \
+      *statep = 0;							      \
 									      \
     put32 (outptr, ch);							      \
     outptr += 4;							      \
-- 
2.55.0

